packages/modular-scripts/src/utils/filterDependencies.ts

```diff
@@ -0,0 +1,37 @@
+import type { Dependency } from '@schemastore/package';
+
+interface FilteredDependencies {
+  external: Dependency;
+  bundled: Dependency;
+}
+
+// Filter out dependencies that are in blocklist
+export function filterDependencies(
+  packageDependencies: Dependency,
+  isApp: boolean,
+): FilteredDependencies {
+  if (isApp) {
+    return {
+      bundled: packageDependencies,
+      external: {},
+    };
+  }
+  const externalBlockList =
+    process.env.EXTERNAL_BLOCK_LIST && !isApp
```

Comment:
Why would we choose to block things? I think exposing this API without an understanding of why this is necessary is probably a bit careless.

Reply:
Mainly for three reasons:
1) If you have an organisation with a private registry, some internal packages could be present in the registry but not in the (possibly public) ESM CDN. In that case, you will need to bundle them.
2) You could be importing a dependency in the monorepo that "looks" like an external dep from the import statement but in reality is resolved by workspace symlinks (and it's not meant to be published to the registry or the CDN). In that case you almost always want to bundle it (we could filter out the workspace dependencies by default, but then you have the inverse problem: what if they're separately published to the CDN and you want to rewrite them?)
3) You could want an escape hatch for when a dependency dependency is broken in the CDN and you can't wait for the CDN to fix it. So you decide to be a bit less efficient and bundle it, until the CDN fixes it and you remove the blocklist and redeploy.

We can *also* have allow lists - they'd be useful in case the CDN exports only a selected range of dependencies (although less useful IMHO). But I wouldn't remove block lists.

Reply:
(or maybe you meant that `EXTERNAL_BLOCK_LIST` is a misnomer? It contains an optional list of dependencies that are to not be rewritten to an external CDN, but bundled in the build file. If you have a better name, happy to change it)